pkg/api/helpers/download.go

```diff
@@ -245,48 +235,107 @@ func (d *Downloader) downloadPresignedPart(ctx context.Context, physicalAddress
 		buf = buf[:partSize]
 	}
 
-	_, err = io.ReadFull(resp.Body, buf)
+	var err error
+	for attempt := 0; attempt <= d.BodyRetries; attempt++ {
+		if attempt > 0 {
+			// sleep for a random time between 0 and 1 second or break on context cancellation
+			select {
+			//nolint:gosec
+			case <-time.After(time.Duration(rand.IntN(DefaultDownloadRetryDelayMs)) * time.Millisecond):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+
+		// create request with range header
+		var req *http.Request
+		req, err = http.NewRequestWithContext(ctx, http.MethodGet, physicalAddress, nil)
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Range", rangeHeader)
+
+		err = executeHTTPRequest(d.HTTPClient, req, func(resp *http.Response) error {
```

Comment:
Aren't you passing the `retryClient` here? It means that it will retry `(attempts)^2` before failing, no?